woodwork/type_sys/utils.py

```diff
@@ -113,7 +113,9 @@ def list_semantic_tags():
     tags_df = tags_df.append(
         pd.DataFrame([['index', False, [ww.type_system.str_to_logical_type(tag) for tag in ['integer', 'double', 'categorical', 'datetime']]],
```

Comment:
We don't ever enforce the valid logical types for `index` that we list here. Can we change this to `Any LogicalType` as well?

Reply:
Seems fine to me